trieste/bayesian_optimizer.py

```diff
@@ -215,3 +221,33 @@ def _save_to_history(
     datasets_copy = {tag: ds for tag, ds in datasets.items()}
     logging_state = LoggingState(datasets_copy, models_copy, copy.deepcopy(acquisition_state))
     history.append(logging_state)
+
+
+def optimize(
+    datasets: Mapping[str, Dataset],
```

Comment:
not sure if it's better to ask for all the data in one argument, or the old data and the new data and let the function merge them